source/scenes/home/Send/Send.tsx

```diff
@@ -188,21 +191,21 @@ const WalletSend: FC<IWalletSend> = ({ initAddress = '', navigation }) => {
 
   const handleAmountChange = useCallback(
     (ev: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
-      let pVal = parseFloat(ev.target.value);
-
-      pVal = isNaN(pVal) ? 0 : pVal;
-
-      // console.log('handleAmountChange', ev.target.value, pVal, amount, (pVal !== Number(amount)))
-
-      if (pVal !== Number(amount)) {
-        setAmount(String(pVal));
+      let valString = ev.target.value;
+      const pVal = parseFloat(ev.target.value);
+      const isValNaN = isNaN(pVal);
+      valString = isValNaN ? '0' : valString;
 
-        setAmountBN(ethers.utils.parseUnits(amount, assetInfo.decimals));
+      const decimalPlaces = valString.includes('.') ? valString.toString().split(".")[1].length || 0 : 0;
+      if (pVal > MAX_AMOUNT_NUMBER || decimalPlaces && decimalPlaces > assetInfo.decimals) return;
 
+      if (valString !== amount) {
+        setAmount(valString);
+        setAmountBN(ethers.utils.parseUnits(valString, assetInfo.decimals));
         estimateGasFee(gasPrice);
       }
       else {
-        setAmount(ev.target.value);
+        setAmount(valString);
```

Comment:
Did you decide against adding any unit tests? This section right here seems like a good candidate to be extracted and tested directly if you do go that route.